web/docs/guides/middleware-customization.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Middleware Customization
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Customizing Express server middleware
+
+Wasp comes with a minimal set of useful Express middleware in every application. While this is good for most users, we realize some may wish to add, modify, or remove some of these choices both globally, or on a per-`api`/path basis.
+
+## Default global middleware
+
+- [Helmet](https://helmetjs.github.io/): Helmet helps you secure your Express apps by setting various HTTP headers. It's not a silver bullet, but it can help!
+- [CORS](https://github.com/expressjs/cors#readme): CORS is a package for providing a middleware that can be used to enable [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) with various options.
+- [Morgan](https://github.com/expressjs/morgan#readme): HTTP request logger middleware.
+- [express.json](https://expressjs.com/en/api.html#express.json) / [body-parser](https://github.com/expressjs/body-parser#bodyparserjsonoptions): Parses incoming request bodies in a middleware before your handlers, making the result available under the `req.body` property. ⚠️ This is required for Wasp Operations to function properly.
+- [express.urlencoded](https://expressjs.com/en/api.html#express.urlencoded) / [body-parser](https://expressjs.com/en/resources/middleware/body-parser.html#bodyparserurlencodedoptions): Returns middleware that only parses urlencoded bodies and only looks at requests where the `Content-Type` header matches the type option.
+- [cookieParser](https://github.com/expressjs/cookie-parser#readme): Parse Cookie header and populate `req.cookies` with an object keyed by the cookie names.
```

Comment:
Cool! O, so which of these they must not touch? You mentioned body-parser is needed for Operations, what about the rest? Are any of these freely removable if they want? I guess Helmet yes, and Morgan also, waht about the rest?

Oh should we drop a NOTE in the code in JS where we actually define these midleware, to update docs if that part is changed, since it is public in a way?

Reply:
Just to close the loop here,  after some testing, only `CORS` and `express.json` appear to be required. Updated docs with warnings for both of those.